baybe/utils/basic.py

```diff
@@ -181,3 +182,75 @@ def find_subclass(base: type, name_or_abbr: str, /):
             f"The class name or abbreviation '{name_or_abbr}' does not refer to any "
             f"of the subclasses of '{base.__name__}'."
         )
+
+
+def register_hooks(
+    target: Callable,
+    pre_hooks: Sequence[Callable] | None = None,
+    post_hooks: Sequence[Callable] | None = None,
+) -> Callable:
+    """Register custom hooks with a given target callable.
+
+    The provided hooks need to be "compatible" with the target in the sense that their
+    signatures are aligned, i.e., no problem arises when arguments intended for the
+    target are passed to the hook:
+
+    * Target and hook need to share the same number of parameters (basic requirement)
```

Comment:
is there any reason we impose this strictly? Seems to me if kwargs from the hook are a subset of kwargs from the target is should still be fine

Reply:
fixed